model/riscv_sys_regs.sail

```diff
@@ -215,6 +215,14 @@ function haveZmmul()  -> bool = true
 /* Zicond extension support */
 function haveZicond() -> bool = true
 
+function legalize_priv_level (p : priv_level) -> priv_level =
+  match (haveUsrMode(), p) {
+    (false,   _) => privLevel_to_bits(Machine),
+    (true, 0b01) => if haveSupMode() then p else privLevel_to_bits(User),
```

Comment:
Is there a specific reason this is legalized to User rather than Machine?

If not this could be simplified to always fall back to Machine if the value is not supported.

Reply:
Agreed choosing lower privilege makes sense - I'd just like to see a comment along the lines of "we are free to choose the legalized privileged, but choosing the lowest valid privilege level is the safest choice".